### PR TITLE
[IT518] CardDAV - phone field synchronization.

### PR DESCRIPTION
**Pull request by arek1-618**
[IT518] CardDAV - phone field synchronization.
thunderbird_2019-03-04_13-29-36
chrome_2019-03-04_13-30-17
chrome_2019-03-04_13-30-48

### CODE REVIEW

File: modules/API/models/CardDAV.php [hunk 1]
```diff
@@ -421,27 +421,28 @@ public function getDavCardsToSync()
...
-				$vcardType = $p->getValue();
-				$vcardType = strtoupper(trim(str_replace('VOICE', '', $vcardType), ','));
-				if ($vcardType == strtoupper($type) && $t->getValue() != '') {
-					\App\Log::trace(__METHOD__ . ' | End | return: ' . $t->getValue());
-
-					return \App\Purifier::purify($t->getValue());
+				$vcardType = strtoupper(trim(str_replace('VOICE', '', $p->getValue()), ','));
+				if ($vcardType === $type && !empty($t->getValue())) {
+					$phone = \App\Purifier::purify($t->getValue());
+					if (!($phone = \App\Fields\Phone::getProperNumber($phone, ($this->user ? $this->user->getId() : null)))) {
```
Comment (rskrzypczak) [inline modules/API/models/CardDAV.php]: ?

File: app/Fields/Phone.php [hunk 2]
```diff
@@ -74,4 +75,29 @@ public static function verifyNumber($phoneNumber, $phoneCountry)
...
+	{
+		if (is_null($userId)) {
+			$userId = \App\User::getCurrentUserId();
+		}
+		$returnVal = false;
+		if (static::getDetails($numberToCheck)) {
+			$returnVal = $numberToCheck;
+		} else {
+			$country = \App\User::getUserModel($userId)->getDetail('default_country_carddav');
+			if (!empty($country) && ($phoneDetails = \App\Fields\Phone::getDetails($numberToCheck, \App\Fields\Country::getCountryCode($country)))) {
```
Comment (rskrzypczak) [inline app/Fields/Phone.php]: ?

File: modules/Users/uitypes/Country.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * UIType country field class.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Adach <a.adach@yetiforce.com>
+ */
+class Users_Country_UIType extends Vtiger_Country_UIType{
```
Comment (rskrzypczak) [inline modules/Users/uitypes/Country.php]: remove

File: layouts/basic/modules/Users/Detail/Field/Country.tpl [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
+{strip}
+<div class="tpl-Users-Detail-Field-Base u-paragraph-m-0">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Users/Detail/Field/Country.tpl]: remove file

File: app/Fields/Country.php [hunk 5]
```diff
@@ -5,6 +5,8 @@
  * @copyright YetiForce Sp. z o.o
  * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
  * @author    Mariusz Krzaczkowski <m.krzaczkowski@yetiforce.com>
+ *
+ * @package   App
```
Comment (mariuszkrzaczkowski) [inline app/Fields/Country.php]:
```php
/**
 * Tools for country class.
 *
 * @package   App
 *
 * @copyright YetiForce Sp. z o.o
 * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
 * @author    Mariusz Krzaczkowski <m.krzaczkowski@yetiforce.com>
 */
```

File: app/Fields/Phone.php [hunk 6]
```diff
@@ -17,15 +17,16 @@ class Phone
...
 	 *
-	 * @param string $phoneNumber
+	 * @param string      $phoneNumber
+	 * @param string|null $phoneCountry
 	 *
-	 * @return bool
+	 * @return bool|array
 	 */
-	public static function getDetails($phoneNumber)
+	public static function getDetails(?string $phoneNumber, ?string $phoneCountry = null)
```
Comment (mariuszkrzaczkowski) [inline app/Fields/Phone.php]:
```suggestion
	public static function getDetails(string $phoneNumber, ?string $phoneCountry = null)
```

File: install/install_schema/scheme.sql [hunk 7]
```diff
@@ -9245,6 +9245,7 @@ CREATE TABLE `vtiger_users` (
   `login_method` varchar(255) DEFAULT 'PLL_PASSWORD',
   `sync_carddav` varchar(100) DEFAULT 'PLL_OWNER',
   `sync_caldav` varchar(100) DEFAULT 'PLL_OWNER',
+  `default_country_carddav` varchar(255) DEFAULT NULL,
```
Comment (mariuszkrzaczkowski) [inline install/install_schema/scheme.sql]:
```suggestion
  `sync_carddav_default_country` varchar(255) DEFAULT NULL,
```

**Pull request merged**